kuma/core/jobs.py

```diff
@@ -10,9 +11,63 @@ class KumaJob(Job):
 
     def key(self, *args, **kwargs):
         key = super(KumaJob, self).key(*args, **kwargs)
-        if self.version is None:
-            return key
-        return '%s#%s' % (key, self.version)
+        if self.version is not None:
+            key = '%s#%s' % (key, self.version)
+        return key
+
+
+class GenerationJob(KumaJob):
+    """
+    A cacheback value job that is part of a generation group.
+
+    The purpose is to refresh several cached values when a generation changes.
+    """
+    generation_age = 60 * 60 * 24 * 365
+
+    def __init__(self, generation_args=None, *args, **kwargs):
+        """
+        Initialize the job and prepare the generation.
+
+        All jobs initialized with the same generation_args list will share the
+        same generation, and all cached values will be invalidated when the
+        generation is invalidated.
+        """
+        self.generation_args = generation_args or []
+        super(KumaJob, self).__init__(*args, **kwargs)
+        self.generation_key = GenerationKeyJob(
+            age=self.generation_age, for_class=self.class_path,
+            generation_args=self.generation_args)
+
+    def key(self, *args, **kwargs):
+        """Create a key that is derived from the generation."""
+        base_key = super(GenerationJob, self).key(*args, **kwargs)
+        gen_key = self.generation_key.key()
+        gen_key_value = self.generation_key.get()
+        return '%s@%s:%s' % (base_key, gen_key, gen_key_value)
+
+    def invalidate_generation(self):
+        """Invalidate the shared generation."""
+        self.generation_key.delete()
+
+
+class GenerationKeyJob(Job):
+    """A generation that is shared by several GenerationJobs."""
+
+    def __init__(self, age, for_class, generation_args, *args, **kwargs):
+        """Initialize but do not create the generation."""
+        super(GenerationKeyJob, self).__init__(*args, **kwargs)
+        self.age = age
+        self.for_class = for_class
+        self.generation_args = generation_args
+
+    def key(self, *args, **kwargs):
+        """Return a key that is derived only from the initial args."""
+        generation_args = ':'.join([str(key) for key in self.generation_args])
```

Comment:
yeah, there are things that could go wrong, but for now the only user is sending a list of one integer.